src/app/Shared/Filter.model.ts

```diff
@@ -1,9 +1,9 @@
-export interface FilterInterface {
-    Erreger: string[];
-    Matrix: string[];
-    Projektname: string[];
-}
+export const mainFilterAttributes = [
+    "Erreger",
+    "Matrix",
+    "Projektname",
+];
 
-export type FilterType = "Erreger" | "Matrix" | "Projektname";
+export type FilterType = typeof mainFilterAttributes[number];
```

Comment:
Or if your intention is to make the filter keys arbitrary:
Remove FilterType and use string instead of FilterType in the other types.
Then move the mainFilterAttributes to some config file.
And finally implement some validation when you read the data to your app. Here you must decide what to do if an attribute is missing in the data (display error or ignore filter)

Reply:
It is planned to get all the information from the server. However, the API is not yet functional. That's why I couldn't adapt my code to it - will be an extra ticket.